wolfcrypt/src/rsa.c

```diff
@@ -3197,11 +3236,12 @@ int wc_RsaSSL_VerifyInline(byte* in, word32 inLen, byte** out, RsaKey* key)
 int wc_RsaSSL_Verify(const byte* in, word32 inLen, byte* out, word32 outLen,
                                                                  RsaKey* key)
 {
-    return wc_RsaSSL_Verify_ex(in, inLen, out, outLen, key , WC_RSA_PKCSV15_PAD);
+    return wc_RsaSSL_Verify_ex(in, inLen, out, outLen, key , WC_RSA_PKCSV15_PAD,
+            WC_HASH_TYPE_NONE);
 }
 
 int  wc_RsaSSL_Verify_ex(const byte* in, word32 inLen, byte* out, word32 outLen,
-                         RsaKey* key, int pad_type)
+                         RsaKey* key, int pad_type, enum wc_HashType hash)
```

Comment:
This will break public API compatibility for `wc_RsaSSL_Verify_ex`. I realize it was added fairly recently, but I believe it was in v4.5.0 release... and we've had support requests for it.. Also no documentation for it.

Reply:
Should I change it to something else like `wc_RsaSSL_Verify_padding`?

Reply:
I dislike continually adding new ex type API's. Consider something more broad if you need to expand this. Also consider there might be another way using an internal API.

Reply:
An additional concern: having `wc_RsaSSL_Verify_ex` with the `pad_type` parameter but without a way to pass in the chosen hash is limiting. If a user wants `WC_RSA_PKCSV15_PAD` padding then they already can use the `wc_RsaSSL_Verify` API. I would argue that the hash parameter is a necessary addition for this API.

Reply:
How does OpenSSL do it? What stops you from doing the same?

Reply:
> How does OpenSSL do it? What stops you from doing the same?

OpenSSL internals have a very different design to wolfSSL. Doing the same is not ideal since the compatibility layer is meant to provide an interface to wolfSSL native functions. We want to keep the compatibility layer as clean as possible.

Reply:
Keep compatibility with openssl. You know best.

Reply:
`wc_RsaSSL_Verify_ex` was left unchanged. I added `wc_RsaSSL_Verify_ex2` which has the necessary parameters for the compatibility layer.